File: app/src/main/java/com/kirchhoff/example/githubclient/repository/GitHubDataSource.java
package com.kirchhoff.example.githubclient.repository;

import android.support.annotation.NonNull;

import com.kirchhoff.example.githubclient.model.Authorization;
import com.kirchhoff.example.githubclient.model.CommitResponse;
import com.kirchhoff.example.githubclient.model.Repository;

import java.util.List;

import rx.Observable;

/**
 * @author Kirchhoff-
 */

public interface GitHubDataSource {

    @NonNull
    Observable<Authorization> auth(@NonNull String login, @NonNull String password);

    @NonNull
    Observable<List<Repository>> repositories();

    @NonNull
    Observable<List<CommitResponse>> getCommits(@NonNull String repos);
}

File: app/src/main/java/com/kirchhoff/example/githubclient/repository/GitHubDataSource.kt
package com.kirchhoff.example.githubclient.repository

/**
 * @author Kirchhoff-
 */


import com.kirchhoff.example.githubclient.model.Authorization
import com.kirchhoff.example.githubclient.model.CommitResponse
import com.kirchhoff.example.githubclient.model.Repository

import rx.Observable

/**
 * @author Kirchhoff-
 */

interface GitHubDataSource {

    fun auth(login: String, password: String): Observable<Authorization>

    fun repositories(): Observable<List<Repository>>

    fun getCommits(repos: String): Observable<List<CommitResponse>>

    fun logout()
}
